### PR TITLE
PEER20-357 create hooks and fix redirect

### PR DESCRIPTION
**Pull request by Dmitrijg24k**
Create useGlobalAdmin and useModeratorRole hooks and fix redirect

### CODE REVIEW

File: app/hooks/useGlobalAdmin.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export const useGlobalAdmin = (redirectPage) => {
+    const [isGlobalAdmin, setGlobalAdmin] = useState(null);
+
+    useEffect(()=>{
+        const permissions = JSON.parse(getCookie('profileinfols') || '""')?.permissions || [];
+
+        setGlobalAdmin(hasGlobalModeratorRole(permissions));
+    }, []); 
+    
+    if (isGlobalAdmin == []) {
```
Comment (Lampardks) [inline app/hooks/useGlobalAdmin.js:18]: this is a wrong comparison. returns a boolean value, use it

so you can't compare
```
isGlobalAdmin == []
```

File: app/hooks/useModeratorRole.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    useEffect(()=>{
+        const permissions = JSON.parse(getCookie('profileinfols') || '""')?.permissions || [];
+
+        setModeratorRole(
+            hasGlobalModeratorRole(permissions) || 
+            hasCommunityModeratorRole(permissions, parseInt(communityId))
+        );
+    }, []); 
+
+    if (isModeratorRole == []) {
```
Comment (Lampardks) [inline app/hooks/useModeratorRole.js:21]: this is a wrong comparison. returns a boolean value, use it

so you can't compare
```
isModeratorRole == []
```

File: app/hooks/useModeratorRole.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, { useState, useEffect } from 'react';
+
+import { getCookie } from '../utils/cookie';
+
+import { hasGlobalModeratorRole, hasCommunityModeratorRole } from '../utils/properties';
+
+import history from '../createdHistory';
+
+export const useModeratorRole = (communityId, redirectPage) => {
```
Comment (Lampardks) [inline app/hooks/useModeratorRole.js:9]: these hooks are almost the same, combine them, example
```
export const useModeratorRole = (redirectPage, communityId = null) => {

setModeratorRole(
            hasGlobalModeratorRole(permissions) || 
            communityId && hasCommunityModeratorRole(permissions, parseInt(communityId))
);
```

File: app/hooks/useGlobalAdmin.js (hunk 1 above)
Comment (Dmitrijg24k) [reply app/hooks/useGlobalAdmin.js]: +

File: app/hooks/useModeratorRole.js (hunk 2 above)
Comment (Dmitrijg24k) [reply app/hooks/useModeratorRole.js]: +

File: app/hooks/useModeratorRole.js (hunk 3 above)
Comment (Dmitrijg24k) [reply app/hooks/useModeratorRole.js]: +

**Review by Lampardks — APPROVED** [review]

**Pull request merged**